### DISCUSSION TITLE
/short.cu/key/sub/path to /expanded.com/foobar/sub/path

### DISCUSSION THREAD

**Discussion by CamielBouchier** [YOURLS/YOURLS#3088] [Q&A]
I would like following functionality:

If "/short.cu/key" expands to "/expanded.com/foobar" then "/short.cu/key/sub/path" should expand to "/expanded.com/foobar/sub/path".

On my local installation I could do it with a ridiculous simple hack: 

In yourls-loader.php:
```
// CB WAS if ( yourls_keyword_is_taken($keyword) or yourls_is_page($keyword) ) {
if ( yourls_keyword_is_taken(explode('/',$keyword)[0]) or yourls_is_page($keyword) ) {
```
in yourls-go.php:
```
// CB
$exploded_keyword = explode('/', $keyword);
// END CB
// CB WAS if( $url = yourls_get_keyword_longurl( $keyword ) ) {
if( $url = yourls_get_keyword_longurl( $exploded_keyword[0] ) ) {
	// CB
	for ($i=1;$i<count($exploded_keyword);$i++) {
		$url = $url . "/" . $exploded_keyword[$i];
	}
	// END CB
    yourls_redirect_shorturl($url, $keyword);
    return;
}
```
Knowing all the warnings against core hacking though ... What would be the equivalent way to get it through plugins?

Camiel B.

**Comment by ozh** [comment]
There are several plugins dealing with passing the query string, ie sho.rt/blah?hey=ho sends to longurlsomething.com/here?hey=ho. They should be a great start for you

**Reply by CamielBouchier** [reply]
@ozh : thanks for the reply. Actually I looked into those as a starting point. And they probably would help in the expansion of the string at the end. The part I didn't find something for is : getting the url in (e.g. "/short.cu/key/some/dir"), which translates internally to something like $key = "key/some/dir". In the database search you want to search for "key" only though. Any hint?

**Reply by ozh** [reply]
What you need is to add a filter to the result of [`yourls_get_request()`](https://github.com/YOURLS/YOURLS/blob/master/includes/functions.php#L899-L957), see if it matches what you're expecting (eg "something slash something slash something", for instance) and deal with it accordingly.
